src/dotnet/CarbonAware.Aggregators/src/CarbonAware/SciScoreAggregator.cs

```diff
@@ -0,0 +1,53 @@
+using CarbonAware.Model;
+using CarbonAware.Interfaces;
+using Microsoft.Extensions.Logging;
+using System.Collections;
+using System.Globalization;
+
+namespace CarbonAware.Aggregators.SciScore
+{
+    public class SciScoreAggregator : ISciScoreAggregator
+    {
+        private readonly ILogger<SciScoreAggregator> _logger;
+
+        private readonly ICarbonIntensityDataSource _carbonIntensityDataSource;
+
+        public SciScoreAggregator(ILogger<SciScoreAggregator> logger, ICarbonIntensityDataSource carbonIntensityDataSource)
+        {
+            this._logger = logger ?? throw new ArgumentNullException(nameof(logger));
+            this._carbonIntensityDataSource = carbonIntensityDataSource;
+        }
+
+        /// <inheritdoc />
+        public async Task<double> CalculateAverageCarbonIntensityAsync(Location location, string timeInterval)
```

Comment:
Not sure where the best place is to document the various exceptions that are expected may be thrown.

Reply:
Added to the interface in #38